### PR TITLE
fix: display relative paths 

### PR DESCRIPTION
**Pull request by Aslemammad**
This tries to display paths as relative paths! I'll address other issues in the next PRs.

### CODE REVIEW

File: src/task.rs [hunk 1]
```diff
@@ -2407,3 +2393,15 @@ impl<'a> Runner<'a> {
...
+
+pub fn relative_path(name: &String) -> String {
+    if name.starts_with(':') || (!name.contains('/') && !name.contains("\\")) {
+        name.to_owned()
+    } else {
+        diff_paths(Path::new(&name), current_dir().unwrap())
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .to_string()
```
Comment (guybedford) [inline src/task.rs:2405]: It should be equivalent and shorter to write:

```suggestion
        diff_paths(Path::new(&name), current_dir().unwrap())
            .unwrap()
            .to_string_lossy();
```

Comment (guybedford) [reply src/task.rs:2405]: Also, since each task can have its own cwd, should we not also explicitly pass the cwd / base?

**Review by guybedford — COMMENTED** [review]
Thanks, definitely a good refactoring.

File: src/task.rs (hunk 1 above)
Comment (Aslemammad) [reply src/task.rs:2405]: Where I can find this cwd / base?

Comment (guybedford) [reply src/task.rs:2405]: `cwd` is a property on all tasks. I think you are already handling this in the task absolute pathing side?

**Comment by guybedford** [comment]
For the Windows test to pass we should replace `\` with `/` I think?

**Comment by Aslemammad** [comment]
Yep, I'll investigate it and work on it on another PR! as far as I think it's a separate issue.

File: src/task.rs [hunk 2]
```diff
@@ -2407,3 +2393,14 @@ impl<'a> Runner<'a> {
         Ok(all_ok)
     }
 }
+
+pub fn relative_path(name: &String, cwd: &String) -> String {
```
Comment (guybedford) [inline src/task.rs:2397]: Better to use an actual string reference I think:

```suggestion
pub fn relative_path(name: &str, cwd: &str) -> String {
```

See eg https://www.ameyalokare.com/rust/2017/10/12/rust-str-vs-String.html

File: src/task.rs [hunk 3]
```diff
@@ -2407,3 +2393,14 @@ impl<'a> Runner<'a> {
...
+    if name.starts_with(':') || (!name.contains('/') && !name.contains("\\")) {
+        name.to_owned()
+    } else {
+        diff_paths(Path::new(&name), Path::new(cwd))
+            .unwrap()
+            .to_string_lossy()
+            .to_string()
```
Comment (guybedford) [inline src/task.rs:2404]: `to_owned` for consistency?

Comment (Aslemammad) [reply src/task.rs:2404]: This won't work because string lossy returns `Cow` type.

Comment (Aslemammad) [reply src/task.rs:2404]: This won't work because string lossy returns `Cow` type.

File: src/task.rs [hunk 4]
```diff
@@ -739,6 +739,7 @@ impl<'a> Runner<'a> {
                 }
                 std::io::stdout().flush().unwrap();
             } else if let Some(cmd_time) = cmd_time {
+                name = relative_path(&name, &self.cwd);
```
Comment (guybedford) [inline src/task.rs:742]: Can we move the `relative_path` call into the `job.display_name` function rather?

**Comment by guybedford** [comment]
Test failing?

**Comment by Aslemammad** [comment]
I rerunned the workflow and it works now!

**Pull request merged**